lnrpc/routerrpc/router.proto

```diff
@@ -58,6 +58,11 @@ message SendPaymentRequest {
     maximum enforced.
     */
     int32 cltv_limit = 9;
+
+    /**
+    Optional route hints to reach the destination through private channels.
+    */
+    repeated lnrpc.RouteHint route_hints = 10 [json_name = "route_hints"];
```

Comment:
Since we already have the payment request, which itself can include routing hints, why do we need a distinct field?

Reply:
I would think we'd want this on the `SendToRoute` request rather than here?

Reply:
In general I think that anything possible via a payment request, should also be possible by specifying the payment parameters separately.

I use this to pre-pay probe a payment for which I have a payment request. I decode the payment request, modify the payment hash to a random value and pass in the separate parameters, including hop hints, to `SendPayment`.

For `SendToRoute` it wouldn't be necessary to pass in hop hints, as those are already part of the supplied route.